tests/kafkatest/services/performance/producer_performance.py

```diff
@@ -62,6 +63,7 @@ def parse_stats(line):
             }
         last = None
         for line in node.account.ssh_capture(cmd):
+            self.maybe_start_jmx_tool(idx, node)
```

Comment:
Calling `maybe_start_jmx_tool` after each line that gets read from the producer process doesn't seem quite right. 

I think we want the behavior to be:
- start producer "asynchronously"
- start jmx tool asynchronously
- wait for producer to finish
- process each line of producer output

I think it would look something like:

```
cmd = "same_as_before &"  # now the cmd is "async"
node.account.ssh(cmd)
wait_until(producer is alive)

self.start_jmx_tool(node)

wait_until(producer is finished)
for line in node.account.ssh_capture("cat /mnt/producer-performance.log"):
  # same as the previous for loop
```